frontend/src/app/pages/settings-page/settings-page.component.ts

```diff
@@ -42,30 +42,49 @@ export class SettingsPageComponent implements OnInit {
             })
     }
 
+    updateUserDetails() {
+        if(!this.validateUserInput()) return;
```

Comment:
Maybe a user should be notified that his/her data update was not successful due to validation failure? I think snack bar would be enough for now :)